### DISCUSSION TITLE
What are the settings in Jukebox Mode?

### DISCUSSION THREAD

**Discussion by MickMorley** [jcorporation/myMPD#972] [Q&A] [answered]
I've read a discussion on what Jukebox mode is but what do these settings do?  For example what is the playlist setting for or the enforce uniqueness?

What is the difference between Song and Album?

Also why does my Jukebox settings say "Playback statistics are disabled?"

I apologize if this explanation is somewhere already. Could you link me to it?

![image](https://user-images.githubusercontent.com/19698105/218225140-a6724c4c-391a-4b51-a64c-9f40642e05e0.png)

**Comment by jcorporation** [comment, marked as answer]
The jukebox mode adds random `songs` or `albums` to the mpd queue if there are less than `min queue length` songs in the mpd queue.

Playlist is the source from which the jukebox adds the songs.

Enforce uniqueness and last player older than are constraints to enforce.

The play statistics warning is displayed if MPD stickers are disabled.

**Reply by MickMorley** [reply]
Thank you, how do you enable stickers?

**Reply by jcorporation** [reply]
This is a mpd setting: https://mpd.readthedocs.io/en/latest/user.html#configuration

**Reply by MickMorley** [reply]
> This is a mpd setting: https://mpd.readthedocs.io/en/latest/user.html#configuration

The setting says that it requires -Dsqlite at compile time.  What I am finding is it should be `-Dsqlite=enabled`.  However, I am using mpd docker containers that are installed via `apk mpd` from the Alpine repository.  I cannot find if they have this enabled. 

Also I have always had the   `sticker_file        "/var/lib/mpd/sticker.sql"` setting but an sql file was never populated.  I have tried 3 different mpd containers with the same result.

**Reply by MickMorley** [reply]
When I run a command like `mpc sticker 'Dave Brubeck/Gone with the Wind - 1959' find rating` the results are

`MPD error: unknown command "sticker"`

**Reply by jcorporation** [reply]
`mpd --version` should show all the compile time settings. 

The jukebox should work without stickers, too. Only the last played since setting does nothing without stickers.

**Reply by MickMorley** [reply]
Here is the results of `mpd --version`

It does work without it, I was curious and wanted to use the sticker feature to further customize the jukebox.

![image](https://user-images.githubusercontent.com/19698105/218763701-48df5693-a328-42bd-9092-02a31e177b3b.png)

**Reply by MickMorley** [reply]
> `mpd --version` should show all the compile time settings.

What setting do I look for to tell if stickers are enabled?

> The jukebox should work without stickers, too. Only the last played since setting does nothing without stickers.

I could of sworn I saw a post where you or someone started Jukebox with settings to ignore disliked songs or something to that effect.

**Reply by jcorporation** [reply]
Right, ignoring dislike songs works only with stickers. Such information is stored in the mpd sticker database.

**Reply by jcorporation** [reply]
You can use `ldd /usr/bin/mpd` to check if mpd is linked with libsqlite.

**Reply by MickMorley** [reply]
> You can use `ldd /usr/bin/mpd` to check if mpd is linked with libsqlite.

I am sorry I did not see this reply. I did update it using `ldd /usr/bin/mpd | sort` but no `libsqlite` was in neither of my 2 mpd containers.  I have 2 that I used to try to find one that has it.

I believe the default `mpd` package for debian and alpine both have sqlite plugin disabled.  It seems rare to have it enabled. 

On my todo list is to learn how to build mpd myself and make an alpine container.

**Reply by jcorporation** [reply]
Compiling MPD is relatively simple. Here is an APKBUILD I use for the MPD in myMPDos. 

https://github.com/jcorporation/myMPDos/tree/master/mympdos/build/mympdos-mpd-stable
